src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -28,11 +30,23 @@ const NavHeader = (props) => {
     isHeaderHide,
     topicDetail,
     setIsFollow,
-    isFollow
+    isFollow,
+    hasSearch,
+    searchText,
+    setSearchText,
+    setDiscoveryLoadingData,
+    isFocus,
+    setIsFocus,
+    fetchDiscoveryData,
+    onCancelToken,
+    placeholderText,
+    setIsFirstTimeOpen,
+    getSearchLayout
   } = props;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are several new props being destructured here. Ensure that these props are being passed in correctly where this component is used, and that they have appropriate default values if they are optional.
```diff
+     isFollow,
+     hasSearch,
+     searchText,
+     setSearchText,
+     setDiscoveryLoadingData,
+     isFocus,
+     setIsFocus,
+     fetchDiscoveryData,
+     onCancelToken,
+     placeholderText,
+     setIsFirstTimeOpen,
+     getSearchLayout
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->